streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -63,83 +93,172 @@ public InternalTopicManager(final StreamsKafkaClient streamsKafkaClient,
-                            " or add more brokers to your cluster.");
+    public void makeReady(final Map<String, InternalTopicConfig> topics) {
+        final Map<String, Integer> existingTopicPartitions = getNumPartitions(topics.keySet(), true);
+        final Set<InternalTopicConfig> topicsToBeCreated = validateTopicPartitions(topics.values(), existingTopicPartitions);
+        if (topicsToBeCreated.size() > 0) {
+            final Set<NewTopic> newTopics = new HashSet<>();
+
+            for (final InternalTopicConfig internalTopicConfig : topicsToBeCreated) {
+                final Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+                final Map<String, String> topicConfig = new HashMap<>(defaultTopicConfigs);
+                for (final String key : topicProperties.stringPropertyNames()) {
+                    topicConfig.put(key, topicProperties.getProperty(key));
+                }
+                newTopics.add(
+                    new NewTopic(
+                        internalTopicConfig.name(),
+                        internalTopicConfig.numberOfPartitions(),
+                        replicationFactor)
+                    .configs(topicConfig));
+            }
+
+            int remainingRetries = retries;
+            boolean retry;
+            do {
+                retry = false;
+
+                final CreateTopicsResult createTopicsResult = adminClient.createTopics(newTopics);
+
+                final Set<String> createTopicNames = new HashSet<>();
+                for (final Map.Entry<String, KafkaFuture<Void>> createTopicResult : createTopicsResult.values().entrySet()) {
+                    try {
+                        createTopicResult.getValue().get();
+                        createTopicNames.add(createTopicResult.getKey());
+                    } catch (final ExecutionException couldNotCreateTopic) {
+                        final Throwable cause = couldNotCreateTopic.getCause();
+                        final String topicName = createTopicResult.getKey();
+
+                        if (cause instanceof TimeoutException) {
+                            retry = true;
+                            log.debug("Could not get number of partitions for topic {} due to timeout. " +
+                                "Will try again (remaining retries {}).", topicName, remainingRetries - 1);
+                        } else if (cause instanceof TopicExistsException) {
+                            createTopicNames.add(createTopicResult.getKey());
+                            log.info(String.format("Topic %s exist already: %s",
+                                topicName,
+                                couldNotCreateTopic.getMessage()));
+                        } else {
+                            throw new StreamsException(String.format("Could not create topic %s.", topicName),
```

Comment:
nit: Could we include `.. due to %{couldNotCreateTopic.name()}` in the message as well?

Reply:
The original exception is passed in as second parameter. Not sure what we gain by adding the Exception class name into the message. Will be included in the stack trace anyway.